### PR TITLE
Event: Warn and fill event aliases

### LINKED ISSUE #230: Warn and fill event shorthands
Core ref https://github.com/jquery/jquery/issues/3214
Docs ref https://github.com/jquery/api.jquery.com/issues/972

### PR DESCRIPTION
**Pull request by dmethvin**
Fixes #230

### CODE REVIEW

File: warnings.md [hunk 1]
```diff
@@ -199,3 +199,15 @@ See jQuery-ui [commit](https://github.com/jquery/jquery-ui/commit/c0093b599fcd58
 **Cause:** The calling code has attempted to attach a `load` event to `window` after the page has already loaded. That means the handler will never run and so is probably not what the caller intended. This can occur when the event attachment is made too late, for example, in a jQuery ready handler. It can also occur when a file is loaded dynamically with jQuery after the page has loaded, for example using the `$.getScript()` method.
 
 **Solution:** If a function `fn` does not actually depend on all page assets being fully loaded, switch to a ready handler `$( fn )` which runs earlier and will aways run `fn` even if the script that contains the code loads long after the page has fully loaded. If `fn` actually does depend on the script being fully loaded, check `document.readyState`. If the value is `"complete"` run the function immediately, otherwise use `$(window).on( "load", fn )`.
+
+### JQMIGRATE: jQuery.fn.click() event shorthand is deprecated
```
Comment (markelog) [inline warnings.md]: Why not generalize this header?

File: src/event.js [hunk 2]
```diff
@@ -75,6 +75,20 @@ jQuery.each( [ "load", "unload", "error" ], function( _, name ) {
...
 
+jQuery.each( ( "blur focus focusin focusout resize scroll click dblclick " +
+	"mousedown mouseup mousemove mouseover mouseout mouseenter mouseleave " +
+	"change select submit keydown keypress keyup contextmenu" ).split( " " ),
+	function( i, name ) {
+
+	// Handle event binding
+	jQuery.fn[ name ] = function( data, fn ) {
+		migrateWarn( "jQuery.fn." + name + "() event shorthand is deprecated" );
+		return arguments.length > 0 ?
```
Comment (markelog) [inline src/event.js]: Can we remember the old functions and call them instead duplicating logic in core?

Comment (mgol) [inline src/event.js]: What do you mean? This code should work even if event shorthands are removed from Core.

Comment (markelog) [inline src/event.js]: As far as i know we don't plan to remove it. And if/when we remove it, warning should be changed regardless

Comment (markelog) [inline src/event.js]: It seems reimplementing is more preferable, judging by the rest of the source anyway

File: warnings.md (hunk 1 above)
Comment (dmethvin) [inline warnings.md]: I tried to figure out the best way to do this. My concern with generalizing the header was that it wouldn't be an exact match. The most common shorthand (I think) is click so i figured this approach would yield the best results with the others in the text. Another way is to duplicate the exact messages but that would get out of hand with so many on the list now. Or I can change it to something like `METHOD`. What do you think?

File: src/event.js (hunk 2 above)
Comment (dmethvin) [inline src/event.js]: I think our approach for 3.0 has been to implement the deprecated functionality even if it's still in core. It does handle the scenario of custom builds better i guess.

**Comment by dmethvin** [comment]
@markelog @mgol LMK if there's anything more you'd like done here.

File: test/event.js [hunk 3]
```diff
@@ -94,6 +94,37 @@ QUnit.test( ".delegate() and .undelegate()", function( assert ) {
...
 
+QUnit.test( "Event aliases", function( assert) {
+	assert.expect( 16 );
+
+	var $div = jQuery( "<div />" );
+
+	"scroll click submit keydown".split( " " ).forEach( function( name ) {
+		expectWarning( "." + name + "()", 1, function() {
+			$div[ name ]( function( event ) {
+				assert.equal( event.type, name, name)
```
Comment (mgol) [inline test/event.js]: A missing space before `)`

Comment (mgol) [inline test/event.js]: ...and a missing semi-colon at the end. :)

File: test/event.js [hunk 4]
```diff
@@ -94,6 +94,37 @@ QUnit.test( ".delegate() and .undelegate()", function( assert ) {
 	} );
 } );
 
+QUnit.test( "Event aliases", function( assert) {
+	assert.expect( 16 );
```
Comment (mgol) [inline test/event.js]: Where does the 16 come from? I counted 14:

* in the first block: 4 warnings & 4 `assert.equal`s
* in the second block: 1 warning & 2 `assert.ok`s
* in the third block: 1 warning & 2 `assert.equal`s

What am I missing?

File: test/event.js (hunk 3 above)
Comment (dmethvin) [inline test/event.js]: Oh great now I have to go figure out why the eslint settings missed that.  :)

**Pull request closed without merging**